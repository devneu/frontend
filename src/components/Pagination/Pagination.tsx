import React from 'react';
import { Pagination as AntdPagination } from 'antd';
import './Pagination.scss'

function itemRender(current: number, type: 'page' | 'next' | 'prev' | 'jump-prev' | 'jump-next', originalElement: {}) {
   if (type === 'prev') {
      return <a>Newer</a>;
   }
   if (type === 'next') {
      return <a>Older</a>;
   }
   return originalElement;
}

export interface PaginationProps {
   defaultCurrentPage: number,
   totalItems: number
}

const Pagination = ({ defaultCurrentPage, totalItems }: PaginationProps) => {
   return (
      <AntdPagination
         defaultCurrent={defaultCurrentPage}
         total={totalItems}
         itemRender={itemRender}
         className="pagination" />
   )
}

export default Pagination;