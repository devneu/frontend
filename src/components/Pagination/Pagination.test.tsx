import Pagination from './Pagination';
import renderer from 'react-test-renderer';

describe('BlogPagination', () => {
   it('Should successfully render', () => {
      const wrapper = renderer
         .create(
            <Pagination
               defaultCurrentPage={3}
               totalItems={500} />
         )
         .toJSON();
      expect(wrapper).toMatchSnapshot();
   });
});