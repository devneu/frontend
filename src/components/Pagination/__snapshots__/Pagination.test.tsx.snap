// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPagination Should successfully render 1`] = `
<ul
  className="ant-pagination pagination"
  style={Object {}}
  unselectable="unselectable"
>
  <li
    aria-disabled={false}
    className="ant-pagination-prev"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex={0}
    title="Previous Page"
  >
    <a
      disabled={false}
    >
      Newer
    </a>
  </li>
  <li
    className="ant-pagination-item ant-pagination-item-1"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex="0"
    title={1}
  >
    <a
      rel="nofollow"
    >
      1
    </a>
  </li>
  <li
    className="ant-pagination-item ant-pagination-item-2"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex="0"
    title={2}
  >
    <a
      rel="nofollow"
    >
      2
    </a>
  </li>
  <li
    className="ant-pagination-item ant-pagination-item-3 ant-pagination-item-active"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex="0"
    title={3}
  >
    <a
      rel="nofollow"
    >
      3
    </a>
  </li>
  <li
    className="ant-pagination-item ant-pagination-item-4"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex="0"
    title={4}
  >
    <a
      rel="nofollow"
    >
      4
    </a>
  </li>
  <li
    className="ant-pagination-item ant-pagination-item-5 ant-pagination-item-before-jump-next"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex="0"
    title={5}
  >
    <a
      rel="nofollow"
    >
      5
    </a>
  </li>
  <li
    className="ant-pagination-jump-next ant-pagination-jump-next-custom-icon"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex="0"
    title="Next 5 Pages"
  >
    <a
      className="ant-pagination-item-link"
    >
      <div
        className="ant-pagination-item-container"
      >
        <span
          aria-label="double-right"
          className="anticon anticon-double-right ant-pagination-item-link-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="double-right"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M533.2 492.3L277.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H188c-6.7 0-10.4 7.7-6.3 12.9L447.1 512 181.7 851.1A7.98 7.98 0 00188 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5zm304 0L581.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H492c-6.7 0-10.4 7.7-6.3 12.9L751.1 512 485.7 851.1A7.98 7.98 0 00492 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5z"
            />
          </svg>
        </span>
        <span
          className="ant-pagination-item-ellipsis"
        >
          •••
        </span>
      </div>
    </a>
  </li>
  <li
    className="ant-pagination-item ant-pagination-item-50"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex="0"
    title={50}
  >
    <a
      rel="nofollow"
    >
      50
    </a>
  </li>
  <li
    aria-disabled={false}
    className="ant-pagination-next"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex={0}
    title="Next Page"
  >
    <a
      disabled={false}
    >
      Older
    </a>
  </li>
  <li
    className="ant-pagination-options"
  >
    <div
      className="ant-select ant-pagination-options-size-changer ant-select-single ant-select-show-arrow"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
    >
      <div
        className="ant-select-selector"
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <span
          className="ant-select-selection-search"
        >
          <input
            aria-activedescendant="undefined_list_0"
            aria-autocomplete="list"
            aria-controls="undefined_list"
            aria-haspopup="listbox"
            aria-owns="undefined_list"
            autoComplete="off"
            className="ant-select-selection-search-input"
            onChange={[Function]}
            onCompositionEnd={[Function]}
            onCompositionStart={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onPaste={[Function]}
            readOnly={true}
            role="combobox"
            style={
              Object {
                "opacity": 0,
              }
            }
            type="search"
            unselectable="on"
            value=""
          />
        </span>
        <span
          className="ant-select-selection-item"
          title="10 / page"
        >
          10 / page
        </span>
      </div>
      <span
        aria-hidden={true}
        className="ant-select-arrow"
        onMouseDown={[Function]}
        style={
          Object {
            "WebkitUserSelect": "none",
            "userSelect": "none",
          }
        }
        unselectable="on"
      >
        <span
          aria-label="down"
          className="anticon anticon-down ant-select-suffix"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </span>
    </div>
  </li>
</ul>
`;
